### DISCUSSION TITLE
Svelte example how to isolate tests each in its own DOM?

### DISCUSSION THREAD

**Discussion by janosh** [vitest-dev/vitest#864] [Q&A]
When using `@testing-library/svelte`, multiple component tests seem to share the same DOM:

```ts
import { cleanup, fireEvent, render } from '@testing-library/svelte'
import { afterEach, expect, test } from 'vitest'
import MyComponent from '../src/lib'

const props = { text: `foobar` }

afterEach(() => cleanup())

test(`test 1`, () => {
  const { getByText } = render(MyComponent, props)
  getByText(`foobar`)
  // ...
})

test(`test 2`, () => {
  const { getByText } = render(MyComponent, props)
  getByText(`foobar`) // error cause finds element from test 1 and this one
  // ...
})
```

Anyone else encounter this and find a good solution?

**Comment by iamyuu** [comment]
You can create your own render function

```js
function renderMyComponent(overrideProps = {}) {
	const stl = render(MyComponent, {
		props: {
			foo: 'bar', // set default props
			...overrideProps,
		},
	});

	return stl;
}
```

**Reply by janosh** [reply]
@iamyuu Not sure how that helps. Could you explain?

**Comment by manuganji** [comment]
I'm working with this library today and this issue still persists. Can't believe this issue has been around since February.

**Reply by manuganji** [reply]
hi @janosh have you found any fix for this?

**Reply by manuganji** [reply]
setting `document.body.innerHTML = ''` in `beforeEach` fixed my problem.

**Reply by manuganji** [reply]
Apparently, jest runner resets the dom only between test files. Here's more about this. 

https://stackoverflow.com/questions/42805128/does-jest-reset-the-jsdom-document-after-every-suite-or-test

I believe Vitest is taking the same approach since the reason is jsdom being expensive to startup.
